### ISSUE TITLE
Switch to flakes, update plutus to something more recent.

### ISSUE THREAD

**Issue by samuelWilliams99** [mlabs-haskell/CardStarter-bulk-airdrop-script#19]
We want to be able to use plutus-extra in this lib, specifically the features listed [here](https://github.com/Liqwid-Labs/plutus-extra/pull/179/files) to replace our local copy. To do this, we need a large plutus update, which will require some restructuring, so we might as well change to flakes as well.

**Comment by samuelWilliams99** [comment]
https://github.com/mlabs-haskell/majority-multisign/

**Comment by samuelWilliams99** [comment]
I've moved the incorporation of plutus-extra part of this task to another ticket, #26 , we can close this once flakify is merged :)
